components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -242,11 +247,24 @@ public void removeExpiredSessionData(Timestamp timestamp) {
             statement = connection.prepareStatement(sqlDeleteTask);
             statement.setTimestamp(1, timestamp);
             statement.execute();
-            connection.commit();
-        } catch (SQLException | IdentityException e) {
-            log.error("Error while removing Session Data.", e);
+            if (!connection.getAutoCommit()) {
+                connection.commit();
+            }
+        } catch (SQLException e) {
+            log.error("Error while removing Session Data ", e);
+        } catch (IdentityException e) {
+            log.error("Error while removing Session Data", e);
         } finally {
-            IdentityDatabaseUtil.closeAllConnections(connection, null, statement);
+            try {
+                if (statement != null) {
```

Comment:
Inside that method, it close one by one the resultset , prepared statement and connection [1]. 
        closeResultSet(rs);
        closeStatement(prepStmt);
        closeConnection(dbConnection);
Each method handle SQLExceptions seperately. For an example check the closeResultSet method [2]. 

[1]. https://github.com/wso2/carbon-identity/blob/master/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityDatabaseUtil.java#L51
[2]. https://github.com/wso2/carbon-identity/blob/master/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityDatabaseUtil.java#L68